src/subplots.ts

```diff
@@ -1634,30 +1657,34 @@ export class BasePlot extends PlotData {
     return new Vertex(mouseDown.x - currentMouse.x, mouseDown.y - currentMouse.y);
   }
 
-  public mouseMove(canvasMouse: Vertex, frameMouse: Vertex): void {
+  public mouseMove(canvasMouse: Vertex, frameMouse: Vertex, absoluteMouse: Vertex): void {
     this.stateUpdate(this.context_show, this.fixedObjects, canvasMouse, 'isHovered', false, false);
-    this.stateUpdate(this.context_show, this.movingObjects, frameMouse, 'isHovered', false, false);
+    this.stateUpdate(this.context_show, this.absoluteObjects, absoluteMouse, 'isHovered', false, false);
+    this.stateUpdate(this.context_show, this.relativeObjects, frameMouse, 'isHovered', false, false);
   }
 
-  public projectMouse(e: MouseEvent): [Vertex, Vertex] {
+  public projectMouse(e: MouseEvent): [Vertex, Vertex, Vertex] {
     const mouseCoords = new Vertex(e.offsetX, e.offsetY);
-    return [mouseCoords.scale(this.initScale), mouseCoords.transform(this.movingMatrix.inverse())]
+    return [mouseCoords.scale(this.initScale), mouseCoords.transform(this.relativeMatrix.inverse()), mouseCoords]
   }
 
-  public mouseDown(canvasMouse: Vertex, frameMouse: Vertex): [Vertex, Vertex, any] {
+  public mouseDown(canvasMouse: Vertex, frameMouse: Vertex, absoluteMouse: Vertex): [Vertex, Vertex, any] {
     let clickedObject: any;
     this.fixedObjects.forEach(object => {if (object.isHovered) {clickedObject = object}})
-    this.movingObjects.forEach(object => {if (object.isHovered) {clickedObject = object}})
+    this.absoluteObjects.forEach(object => {if (object.isHovered) {clickedObject = object}})
+    this.relativeObjects.forEach(object => {if (object.isHovered) {clickedObject = object}})
     if (this.fixedObjects.indexOf(clickedObject) != -1) {clickedObject.mouseDown(canvasMouse)}
-    if (this.movingObjects.indexOf(clickedObject) != -1) {clickedObject.mouseDown(frameMouse)}
+    if (this.absoluteObjects.indexOf(clickedObject) != -1) {clickedObject.mouseDown(absoluteMouse)}
+    if (this.relativeObjects.indexOf(clickedObject) != -1) {clickedObject.mouseDown(frameMouse)}
```

Comment:
I am puzzled by the fact that we often need to do exactly the same stuff (or almost) to absolute and relativeObjects. This often mark a structure that should be OO.

Relative and Absolute objects would inherit from an object and specify their own way of doing these

Reply:
You're right and i've done it

Reply:
Maybe not in an optimized way but it introduces a fine way to do what has to be done